app/controllers/proposal/approved_proposals_controller.rb

```diff
@@ -0,0 +1,47 @@
+module Proposal
+  class ApprovedProposalsController < ManageMetadataController
+    # There are exceptions in this code for proposal_mode checks for development
+    # and test to prevent the need to run two servers while using those
+    # environments.
+
+    skip_before_action :ensure_user_is_logged_in
+    skip_before_action :refresh_urs_if_needed, :refresh_launchpad_if_needed, :provider_set?, :proposal_approver_permissions
+
+    def approved_proposals
+      token_and_client_id = request.headers.fetch('Echo-Token', ':').split(':')
+      passed_token = token_and_client_id[0]
+      passed_client_id = token_and_client_id[1]
+
```

Comment:
You can say passed_token, passed_client_id = ...
so you don't need to intermediate variable

Reply:
I thought that a, b = ['a', 'b', 'c'] would yield a = 'a' and b = ['b', 'c'], but I was mistaken.  Changed.